pkg/restapi/kms/operation/operations.go

```diff
@@ -219,6 +223,57 @@ func sign(provider *kmsProvider, keystoreID, keyID, message string) ([]byte, err
 	return signature, nil
 }
 
+func (o *Operation) verifyHandler(rw http.ResponseWriter, req *http.Request) {
+	var request verifyReq
+	if err := json.NewDecoder(req.Body).Decode(&request); err != nil {
+		writeErrorResponse(rw, http.StatusBadRequest, fmt.Sprintf(receivedBadRequest, err))
+		return
+	}
+
+	sig, err := base64.URLEncoding.DecodeString(request.Signature)
+	if err != nil {
+		writeErrorResponse(rw, http.StatusBadRequest, fmt.Sprintf(receivedBadRequest, err))
+		return
+	}
+
+	keystoreID := mux.Vars(req)["keystoreID"]
+	keyID := mux.Vars(req)["keyID"]
+
+	provider, err := prepareKMSProvider(o.provider, KMSCreatorContext{
+		KeystoreID: keystoreID,
+		Passphrase: request.Passphrase,
+	})
+	if err != nil {
+		writeErrorResponse(rw, http.StatusInternalServerError, fmt.Sprintf(createKMSProviderFailure, err))
+		return
+	}
+
+	err = verify(provider, keystoreID, keyID, sig, request.Message)
+	if err != nil {
+		if errors.Is(err, kmsservice.ErrInvalidSignature) {
+			rw.WriteHeader(http.StatusOK)
```

Comment:
Do you really mean to write `http.StatusOK` here?

Reply:
(and if so - it can be omitted since it's written automatically)

Reply:
I intentionally left that `rw.WriteHeader(http.StatusOK)` here to denote that despite we are returning the error message the response itself is considered as not failed. The other similar case near the end of the function doesn't write explicitly the StatusOK header because it's clear that it's a happy path.

Reply:
Makes sense :)